fenix/app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -719,6 +719,12 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
 
             searchWidgetInstalled.set(settings.searchWidgetInstalled)
 
+            if (settings.sharedPrefsUUID.isEmpty()) {
+                settings.sharedPrefsUUID = sharedPrefsUuid.generateAndSet().toString()
+            } else {
```

Comment:
Do we need else? Should `sharedPrefUuid.set()` always called

Reply:
We want to see if the value changes over time. I set the `lifetime` for the new `sharedPrefsUuid` Glean metric as `ping`, so reading the value from shared prefs and setting that for the metric makes sense to me.

Does that sound reasonable to you?

Reply:
Yes, as discussed `generateAndSet()` does also set the metrics value for the ping.  Your implementation here is correct.